File: libs/python/flow_like/sinks.py
from __future__ import annotations

from typing import Any

from ._http import HTTPClient


class SinksMixin(HTTPClient):
    def trigger_http_sink(
        self,
        app_id: str,
        path: str,
        method: str = "POST",
        body: Any = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        resp = self._request(
            method.upper(),
            f"/sink/trigger/http/{app_id}/{path}",
            json=body,
            **kwargs,
        )
        return resp.json()

    async def atrigger_http_sink(
        self,
        app_id: str,
        path: str,
        method: str = "POST",
        body: Any = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        resp = await self._arequest(
            method.upper(),
            f"/sink/trigger/http/{app_id}/{path}",
            json=body,
            **kwargs,
        )
        return resp.json()


__all__ = ["SinksMixin"]

File: libs/platform/nodejs/src/sinks.ts
import type { HttpClient } from "./client.js";
import type { HttpSinkOptions } from "./types.js";

export function createSinkMethods(http: HttpClient) {
	return {
		async triggerHttpSink(
			appId: string,
			path: string,
			method = "POST",
			body?: unknown,
			options?: HttpSinkOptions,
		): Promise<unknown> {
			return http.request(method, `/sink/trigger/http/${appId}/${path}`, {
				body,
				headers: options?.headers,
				signal: options?.signal,
			});
		},
	};
}
